pkg/agent/controller/agent.go

```diff
@@ -497,3 +528,25 @@ func getGlobalIpFromService(service *corev1.Service) string {
 
 	return ""
 }
+
+func (e *Controller) remoteEndpointSliceToLocal(obj runtime.Object, op syncer.Operation) (runtime.Object, bool) {
+	endPointSlice := obj.(*discovery.EndpointSlice)
+
+	newEndPointSlice := newEndPointSlice(endPointSlice)
+
+	return newEndPointSlice, false
+}
+
+func newEndPointSlice(endpointSlice *discovery.EndpointSlice) *discovery.EndpointSlice {
+	return &discovery.EndpointSlice{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      endpointSlice.Name,
+			Namespace: endpointSlice.Namespace,
```

Comment:
This needs to be original source namespace stored in the labels.

Reply:
Done.